### PR TITLE
shadow dom is optional via attribute presence

### PR DESCRIPTION
**Pull request by argyleink**
made an api for what was optional via commented code 👍

### CODE REVIEW

**Comment by notwaldorf** [comment]
This is cool! Unfortunately this reverses the _current_ behaviour of the element, thus forcing everyone to update (if they want to maintain the status quo). What to you think about naming renaming the attribute to `as-decorator`, and reversing the if statement? (if `as-decorator` is present, then don't use a shadow root)

**Comment by argyleink** [comment]
works for me! you'll find the PR updated for your review

File: lazy-image.js [hunk 1]
```diff
@@ -27,14 +27,17 @@ class LazyImage extends HTMLElement {
...
 
-    // Add the <img> element inside the shadow root.
-    const shadow = this.attachShadow({mode: 'open'});
-    shadow.appendChild(this._img);
-
-    // Note: If you don't want to use the shadow DOM, you could've done
-    // this instead of the above 2 lines:
-    // this.appendChild(this._img);
+  connectedCallback() {
+    if (!this.hasAttribute('shadow')) {
```
Comment (notwaldorf) [inline lazy-image.js]: This is the same as your previous change (the user still has to put the `shadow` attribute for the status quo to work. I meant:
```
if (this.hasAttribute('as-decorator') {
  this.appendChild(this._img);
} else {
  const shadow = this.attachShadow({mode: 'open'});
  shadow.appendChild(this._img);
}
```

**Review by notwaldorf — CHANGES_REQUESTED** [review]

File: lazy-image.js (hunk 1 above)
Comment (argyleink) [reply lazy-image.js]: you're right! i only did half the changes requested.. i should have spent the time in my updated PR earlier, i totally rushed it and forgot half the comments you made. sorry! will update shortly!

**Review by argyleink — COMMENTED** [review]
ready for review!

**Review by notwaldorf — APPROVED** [review]

**Comment by notwaldorf** [comment]
Perfect, thanks!

**Pull request merged**